docs/standard/serialization/system-text-json/polymorphism.md

```diff
@@ -136,7 +139,607 @@ The following example shows the JSON that results from the preceding code:
 ```
 
 > [!NOTE]
-> This article is about serialization, not deserialization. Polymorphic deserialization is not supported, but as a workaround you can write a custom converter, such as the example in [Support polymorphic deserialization](converters-how-to.md#support-polymorphic-deserialization).
+> This article is about serialization, not deserialization. Polymorphic deserialization is not supported, but as a workaround you can write a custom converter, such as the example in [Support polymorphic deserialization](converters-how-to.md#support-polymorphic-deserialization). .NET 7 supports polymorphic serialization and deserialization, for more information see [How to serialize properties of derived classes with System.Text.Json in .NET 7](polymorphism.md?pivots=dotnet-7-0).
+
+:::zone-end
+:::zone pivot="dotnet-7-0"
+
+[!INCLUDE [preview-content](../../../../includes/preview-content.md)]
+
+Beginning with .NET 7, `System.Text.Json` supports polymorphic type hierarchy serialization and deserialization with attribute annotations.
+
+| Attribute | Description |
+|--|--|
+| <xref:System.Text.Json.Serialization.JsonDerivedTypeAttribute> | When placed on a type declaration, indicates that the specified subtype should be opted into polymorphic serialization. Exposes the ability to specify a type discriminator. |
+| <xref:System.Text.Json.Serialization.JsonPolymorphicAttribute> | When placed on a type, indicates that the type should be serialized polymorphically. Exposes various options to configure polymorphism serialization and deserialization on the target type. |
+
+For example, suppose you have a `WeatherForecastBase` class and a derived class `WeatherForecastWithCity`:
+
+:::code language="csharp" source="snippets/system-text-json-how-to/csharp/WeatherForecast.cs" id="WFB":::
+:::code language="vb" source="snippets/system-text-json-how-to/vb/WeatherForecast.vb" id="WFB":::
+
+:::code language="csharp" source="snippets/system-text-json-how-to/csharp/WeatherForecast.cs" id="WFWC":::
+:::code language="vb" source="snippets/system-text-json-how-to/vb/WeatherForecast.vb" id="WFWC":::
+
+And suppose the type argument of the `Serialize` method at compile time is `WeatherForecastBase`:
+
+:::code language="csharp" source="snippets/system-text-json-how-to/csharp/SerializePolymorphic.cs" id="SerializeWithAttributeAnnotations":::
+:::code language="vb" source="snippets/system-text-json-how-to/vb/SerializePolymorphic.vb" id="SerializeWithAttributeAnnotations":::
+
+In this scenario, the `City` property is serialized even though the `weatherForecastBase` object is actually a `WeatherForecastWithCity` object. This configuration enables polymorphic serialization for `WeatherForecastBase`, specifically when the runtime type is `WeatherForecastWithCity`:
```

Comment:
```suggestion
In this scenario, the `City` property is serialized because the `weatherForecastBase` object is actually a `WeatherForecastWithCity` object. This configuration enables polymorphic serialization for `WeatherForecastBase`, specifically when the runtime type is `WeatherForecastWithCity`:
```

Reply:
I also don't see where `weatherForecastBase` is declared in this example.